### ISSUE TITLE
not able to play item on PlexMediaPlayer?

### ISSUE THREAD

**Issue by derailius** [pkkid/python-plexapi#395]
I'm sorry if this is covered somewhere, i've searched and searched, and cannot find a reason as to why.

# Example 4: Play the movie Cars on another client.
# Note: Client must be on same network as server.
cars = plex.library.section('Movies').get('Cars')
client = plex.client("Michael's iPhone")
client.playMedia(cars)

(Obviously I have Cars in my library, and I've changed the plex.client accordingly)
this works just fine to play on my Roku devices, and my iPhone app, even OpenPHT,  but not Plex Media Player?

I get:

DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /clients HTTP/1.1" 200 460
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): 192.168.0.23:32433
DEBUG:urllib3.connectionpool:http://192.168.0.23:32433 "GET /resources HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library HTTP/1.1" 200 289
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library/sections HTTP/1.1" 200 1611
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library/sections/2/all HTTP/1.1" 200 298041
DEBUG:urllib3.connectionpool:Resetting dropped connection: 192.168.0.23
DEBUG:urllib3.connectionpool:http://192.168.0.23:32433 "GET /player/timeline/poll?wait=0&commandID=1 HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Resetting dropped connection: 192.168.0.23
DEBUG:urllib3.connectionpool:http://192.168.0.23:32433 "GET /player/timeline/subscribe?commandID=2&port=32400&protocol=http HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library/metadata/200471?checkFiles=1&includeExtras=1&includeRelated=1&includeOnDeck=1&includeChapters=1&includePopularLeaves=1&includeConcerts=1&includePreferences=1 HTTP/1.1" 200 30846
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "POST /playQueues?includeChapters=1&includeRelated=1&key=/library/metadata/200471&repeat=0&shuffle=0&type=video&uri=library%3A//XXXXXXX/item//library/metadata/200471 HTTP/1.1" 200 30442
DEBUG:urllib3.connectionpool:Resetting dropped connection: 192.168.0.23
Traceback (most recent call last):
  File "test.py", line 20, in <module>
    client.playMedia(always)
  File "/home/derailius/.local/lib/python2.7/site-packages/plexapi/client.py", line 480, in playMedia
    }, **params))
  File "/home/derailius/.local/lib/python2.7/site-packages/plexapi/client.py", line 207, in sendCommand
    return self.query(key, headers=headers)
  File "/home/derailius/.local/lib/python2.7/site-packages/plexapi/client.py", line 161, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "/home/derailius/.local/lib/python2.7/site-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/home/derailius/.local/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/derailius/.local/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/derailius/.local/lib/python2.7/site-packages/requests/adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='192.168.0.23', port=32433): Read timed out. (read timeout=30)

and then after a while Plex Media Player crashes.  I've tried different versions of PMP and on different computers. I know there isn't a firewall issue or anything because OpenPHT works just fine.
in addition, the NEW Plex player doesn't work either.  My preferred player is Plex Media Player.
I'm not sure how to proceed, any help would be appreciated.

**Comment by blacktwin** [comment]
I know this is an older issue but I was able to play without issue on the following build:
```
Windows 10 or later (64-bit only).
1.4.1.940-574c2fa7
November 21, 2019
```
Using the following:
cars = plex.library.section('Movies').get('Cars')
client = plex.client("ComptuerName")
client.playMedia(cars)
```

I did test with a older version of PMP and I could not identify the player in the list of available clients from `plex.clients()` which may have been your issue @derailius. All the available clients were listed as chrome. Can you please try this again and confirm that it is working for you?

**Comment by derailius** [comment]
Okay, apparently what you're using is Plex Player, and not Plex Media Player. What you've tested is the media center portion of Plex.

This is what i use on Windows 10. 

https://forums.plex.tv/t/plex-media-player/120475/91

**Comment by blacktwin** [comment]
Yes, that's right. Ok, I'll try the version you linked to confirm. I believe I had that installed before but I'm pretty sure it wasn't up to date and it didn't run its usual auto update at launch.

**Comment by Hellowlol** [comment]
I did some changes to regarding the playback some time ago. Can you try using fresh pull from GitHub?

**Comment by derailius** [comment]
fresh pull

`DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library/sections HTTP/1.1" 200 1601
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /library/metadata/162550 HTTP/1.1" 200 7578
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): 192.168.0.23:32433
DEBUG:urllib3.connectionpool:http://192.168.0.23:32433 "GET /player/timeline/poll?wait=0&commandID=1 HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Resetting dropped connection: 192.168.0.23
DEBUG:urllib3.connectionpool:http://192.168.0.23:32433 "GET /player/timeline/subscribe?commandID=2&port=32400&protocol=http HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "POST /playQueues?includeChapters=1&includeRelated=1&key=/library/metadata/162550&repeat=0&shuffle=0&type=video&uri=library%3A//e70dbcf2-ff2d-4a34-b1b2-eee859db0019/item//library/metadata/162550 HTTP/1.1" 200 41621
DEBUG:urllib3.connectionpool:Resetting dropped connection: 192.168.0.23
Traceback (most recent call last):
  File "playit2.py", line 28, in <module>
    client.playMedia(movie)
  File "/var/lib/plexmediaserver/.local/lib/python2.7/site-packages/plexapi/client.py", line 486, in playMedia
    }, **params))
  File "/var/lib/plexmediaserver/.local/lib/python2.7/site-packages/plexapi/client.py", line 204, in sendCommand
    return query(key, headers=headers)
  File "/var/lib/plexmediaserver/.local/lib/python2.7/site-packages/plexapi/client.py", line 161, in query
    response = method(url, headers=headers, timeout=timeout, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='192.168.0.23', port=32433): Read timed out. (read timeout=30)
`

**Comment by Hellowlol** [comment]
What mode do you use in pmp when this happens and what version os are you using?

**Comment by derailius** [comment]
It does this on TV Mode and Web mode. PMP version 2.47.0.1035-e74d31b  Windows 10 Pro 1909 OS Build 18363.535  If i use OpenPHT on either of the two computers i've tried it on, it works fine.  If i use my iphone it works fine, it just doesn't like PMP for some reason.

**Comment by derailius** [comment]
I just tested this on the Plex Player for Windows app and it too doesn't work.  

Plex for Windows

Windows 10 or later (64-bit only).

1.4.1.940-574c2fa7

November 21, 2019


But, with fresh pull I tested it on OpenPHT and iPhone, no problem

**Comment by blacktwin** [comment]
Any reason why you're holding on to an older version?

`Plex-1.4.1.940-574c2fa7-x86_64` worked for me. This is the newest release of the Plex Player.
`PlexMediaPlayer-3.104.2` did not work for me. This had been previously installed.
For the PMP 3.104.2 version, it does not show up in the `plex.clients()` list when open or playing. I also removed it from the Authorized Device List and signed back in just to make sure. It would seem that with newer releases older versions get dropped from the clients listing for some reason. 

@derailius Can you show your output of this when you have your version of PMP open.
```python
for client in plex.clients():
    print(client.title)
    # print(vars(client)) # for more information
```
And provide the code you're running.

**Comment by derailius** [comment]
I'm not holding on to an older version, lol I'm using the latest Plex Media Player (2.47.0.1035-e74d31b) which didn't work and I tried the Plex Player (Plex-1.4.1.940-574c2fa7-x86_64) which also did not work.
There is a BETA of Plex Media Player 3.0.0.24 that i didn't even try.

Plex really has a mess with that, it creates too much confusion when they have players named pretty much the same.

DEBUG:urllib3.connectionpool:http://192.168.0.110:32400 "GET /clients HTTP/1.1" 200 371
Bryan-PC  <---the computer with Plex Media Player 2.47.0.1035-e74d31b running.

from plexapi.server import PlexServer
import sys

baseurl = 'http://192.168.0.110:32400'
token = 'TOKEN'
plex = PlexServer(baseurl, token)
proxyThroughServer = False
includeToken=True

import logging
logging.basicConfig(level=logging.DEBUG)

for client in plex.clients():
    print(client.title)

client = plex.client("Bryan-PC")
movie = plex.library.section('Movies').fetchItem(162550)
client.playMedia(movie)

**Comment by derailius** [comment]
ok, this is odd, Plex Player isn't showing the correct IP or port

<Server name="Toshiba" host="127.0.0.1" address="127.0.0.1" port="32400" machineIdentifier="IDENT" version="1.4.1.940-574c2fa7" protocol="plex" product="Plex for Windows" deviceClass="pc" protocolVersion="3" protocolCapabilities="timeline,playback,navigation,mirror,playqueues"/>

But Plex Media Player and OpenPHT does
<Server name="Bryan-PC" host="192.168.0.23" address="192.168.0.23" port="32433" machineIdentifier="IDENT" version="2.47.0.1035-e74d341b" protocol="plex" product="Plex Media Player" deviceClass="pc" protocolVersion="1" protocolCapabilities="navigation,playback,timeline,mirror,playqueues"/>

<Server name="htpc" host="192.168.0.100" address="192.168.0.100" port="3005" machineIdentifier="IDENT" version="1.8.0.148-573b6d73" protocol="plex" product="OpenPHT" deviceClass="pc" protocolVersion="1" protocolCapabilities="navigation,playback,timeline,mirror,playqueues"/>

<Server name="Bryan's iPhone" host="192.168.0.194" address="192.168.0.194" port="32500" machineIdentifier="IDENT" version="6.5.2" protocol="plex" product="Plex for iOS" deviceClass="phone" protocolVersion="2" protocolCapabilities="playback,playqueues,timeline,provider-playback"/>

so to recap, the only player on this list that i tested that works with the script i posted above is OpenPHT and the Plex app on iPhone

**Comment by Hellowlol** [comment]
I just tried with Plex Media Player 2.47 (the same version ad you) and i can start playback on while in normal and tv mode. both directly and by enabling ```client.proxyThroughServer()``` check that Advertise as player
Allow other Plex apps to fling content to this device and control it remotely. is checked while you are in normal mode,

**Comment by derailius** [comment]
are you using the Mac version? Please excuse my limited knowledge of python, is this right? If so I tried it and Advertise as a player was also already enabled.

```python
from plexapi.server import PlexServer

baseurl = 'http://192.168.0.110:32400'
token = 'TOKEN'
plex = PlexServer(baseurl, token)
proxyThroughServer = True
includeToken=True
import logging
logging.basicConfig(level=logging.DEBUG)

for client in plex.clients():
    print(client.title)

client = plex.client("Bryan-PC")

movie = plex.library.section('Movies').fetchItem(162550)
client.proxyThroughServer()
client.playMedia(movie)

**Comment by Hellowlol** [comment]
Im using the windows version. Yeah, look correct but  
````proxyThroughServer = True
includeToken=True```` isnt needed.

**Comment by derailius** [comment]
well, I guess I give up then, because I've tried that and no matter what computer i host PMP on, it refuses to play. just times out then eventually PMP crashes. I appreciate all of your help.

**Comment by Hellowlol** [comment]
Only thing i can advice you to is to dig in the log and see if you find anything there.

**Comment by amazingr4b** [comment]
@derailius out of curiosity, if you look at the client.py file in your site-packages/plexapi directory, does it have line 494 as "'type': mediatype,"?

**Comment by derailius** [comment]
> 
> 
> @derailius out of curiosity, if you look at the client.py file in your site-packages/plexapi directory, does it have line 494 as "'type': mediatype,"?

no, line 494 is:

`repeat (int): Repeat mode (0=off, 1=repeatone, 2=repeatall; optional).`

and the only mention of mediatype in client.py is:

`      for mediatype in self.timeline():
            if mediatype.get('state') == 'playing':
                return True
            if includePaused and mediatype.get('state') == 'paused':
                return True
        return False
`
which are lines 541 - 546

**Comment by amazingr4b** [comment]
Sounds like you might have the same problem I did, which was I did not have the most recent client.py file after doing a pip install. The one I got from pip does not include the december fixes. After manually grabbing the current one on github and using that I've been able to control PMP and the xbox1 clients without getting the timeout or connection refused errors.

**Comment by pergolafabio** [comment]
hi @amazingr4b  where can i find that client.py from december? i see last update in november?

**Comment by blacktwin** [comment]
@derailius is this issue still active?
